maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -242,11 +242,12 @@
     private String[] additionalClasspathElements;
 
     /**
-     * The test source directory containing test class sources.
+     * The test source directory containing test class sources. Only used by the TestNG provider to process JavaDoc
+     * tags.
      *
      * @since 2.2
      */
-    @Parameter( defaultValue = "${project.build.testSourceDirectory}", required = true )
+    @Parameter( defaultValue = "${project.build.testSourceDirectory}", readonly = true )
```

Comment:
same as above

Reply:
The problem I'm having with this being marked as required is, that is will make `testSourceDirectory` show up at the very top of the [documentation page](https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html). This unnecessary highlights a property that is only used by legacy implementations of TestNG. As said above, I would remove it completely.